zerver/lib/actions.py

```diff
@@ -1338,13 +1338,7 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
             sender_id=message['message'].sender_id,
             stream_topic=stream_topic,
             possibly_mentioned_user_ids=mention_data.get_user_ids(),
-            # TODO: We should improve the `mention_data` logic to
-            # populate the possible_wildcard_mention field based on
-            # whether wildcard mention syntax actually appears in the
-            # message, to avoid wasting resources computing
-            # wildcard_mention_user_ids for messages that could
-            # not possibly contain a wildcard mention.
-            possible_wildcard_mention=True,
+            possible_wildcard_mention=mention_data.message_has_wildcards(),
```

Comment:
This change should also be made in the message update code path; do that in a new commit because we currently don't use `mention_data` at all there, which we should fix as part of this.

I think a reasonable way to manually test this is to make `mention_data.message_has_wildcard` return always `False` or always `True` and confirm that both cases fail other mention-delivery tests (I'd expect `test_event_queue` to fail).  If one of those doesn't, we should look into that.

Reply:
Actually, I think I expect always True to work fine -- it'll just waste resources :).  So we'll count on the unit tests to handle that piece; we should just make sure tests fail with `False`.

Reply:
`test_event_queue` fails with the always False approach. I'll add this change to the update_message flow as well.

Reply:
Great!